plugins/obs-qsv11/QSV_Encoder.cpp

```diff
@@ -144,27 +147,40 @@ bool prefer_current_or_igpu_enc(int *iGPUIndex)
 			    (AdapterDesc.AdapterLuid.LowPart == luid.LowPart) &&
 			    (AdapterDesc.AdapterLuid.HighPart ==
 			     luid.HighPart)) {
-				hasCurrent = true;
-				*iGPUIndex = adapterIndex;
-				break;
+				adapter_info->hasCurrent = true;
+				adapter_info->currentGPUIndex = adapterIndex;
 			}
 
 			if (AdapterDesc.DedicatedVideoMemory <=
 			    512 * 1024 * 1024) {
-				hasIGPU = true;
-				if (iGPUIndex != NULL) {
-					*iGPUIndex = adapterIndex;
-				}
+				adapter_info->hasIGPU = true;
+				adapter_info->iGPUIndex = adapterIndex;
 			} else {
-				hasDGPU = true;
+				adapter_info->hasDGPU = true;
+				adapter_info->dGPUIndex = adapterIndex;
 			}
 		}
 	}
 
 	pFactory->Release();
 	FreeLibrary(hDXGI);
 
-	return hasCurrent || (hasIGPU && hasDGPU);
+	return true;
+}
+
+bool qsv_av1_supported(mfxIMPL impl)
+{
+	bool av1_supported = false;
+
+	QSV_Encoder_Internal *pEncoder = new QSV_Encoder_Internal(impl, ver);
+
+	av1_supported = pEncoder->AV1_Supported(impl);
+
+	delete pEncoder;
+	if (pEncoder)
+		is_active.store(false);
```

Comment:
What exactly is the function of this `delete <x>; if(x)` pattern? Also `is_active` doesn't seem to ever be read?